tools/submission/submission-checker.py

```diff
@@ -0,0 +1,554 @@
+            if float(acc) >= a:
+                is_valid = True
+                break
+        if not is_valid:
+            log.error("%s accuracy not met: expected=%f, found=%f", dir, acc_target, acc)
+
+    # check if there are any errors in the detailed log
+    fname = os.path.join(dir, "mlperf_log_detail.txt")
+    if not os.path.exists(fname):
+        log.error("%s is missing", fname)
+        is_valid = False
+    else:
+        with open(fname, "r") as f:
+            for line in f:
+                # look for: ERROR
+                if "ERROR" in line:
+                    if ignore_errors(line):
+                        continue
+                    # TODO: should this be a failed run?
+                    log.error("%s contains error: %s", fname, line)
+                    is_valid = False
+    return is_valid, acc
+
+
+def check_performance_dir(config, model, dir):
+    is_valid = False
+    rt = {}
+    # look for: Result is: VALID
+    fname = os.path.join(dir, "mlperf_log_summary.txt")
+    with open(fname, "r") as f:
+        for line in f:
+            m = re.match(r"^Result\s+is\s*\:\s+VALID", line)
+            if m:
+                is_valid = True
+            m = re.match(r"^\s*([\w\s.\(\)\/]+)\s*\:\s*([\w\+\.]+).*", line)
+            if m:
+                rt[m.group(1).strip()] = m.group(2).strip()
+
+    model = model_map(config, model)
+    performance_sample_count = config.get_performance_sample_count(model)
+    if int(rt['performance_sample_count']) < performance_sample_count:
+        log.error("%s performance_sample_count should be %d", fname, performance_sample_count)
+        is_valid = False
+
+    # check if there are any errors in the detailed log
+    fname = os.path.join(dir, "mlperf_log_detail.txt")
+    with open(fname, "r") as f:
+        for line in f:
+            # look for: ERROR
+            if "ERROR" in line:
+                if ignore_errors(line):
+                    continue
+                log.error("%s contains error: %s", fname, line)
+                is_valid = False
+
+        for seed in ["qsl_rng_seed", "sample_index_rng_seed", "schedule_rng_seed"]:
+            if int(rt[seed]) != config.seeds[seed]:
+                log.error("%s %s is wrong, expected=%s, found=%s", fname, seed, config.seeds[seed], rt[seed])
+
+    scenario = rt["Scenario"]
+    res = float(rt[RESULT_FIELD[scenario]])
+    if scenario in ["Single Stream"]:
+        res /= TO_MS
+
+    return is_valid, res
+
+
+def files_diff(list1, list2):
+    """returns a list of files that are missing or added."""
+    if list1 and list2:
+        for i in ["mlperf_log_trace.json", "results.json"]:
+            try:
+                list1.remove(i)
+            except:
+                pass
+        if len(list1) > len(list2):
+            return list(set(list1) - set(list2))
+        else:
+            return list(set(list2) - set(list1))
+    return []
+
+
+def check_results_dir(config, dir, filter_submitter, csv):
```

Comment:
Could you add a comment showing the example valid directory structure? Might be easier for users to look at a figure than to read through the codes. Thanks

Reply:
done.